pynitrokey/cli/nk3/test.py

```diff
@@ -233,6 +237,191 @@ def select(conn: CardConnection, aid: list[int]) -> bool:
+    "RecreationDeleteUserId",
+    "RecreationRecreateUserId",
+    "RecreationCreateSession",
+    "RecreationAuthSession",
+    "RecreationDeleteAttack",
+    "Rsa2048Gen",
+    "Rsa2048Sign",
+    "Rsa2048Verify",
+    "Rsa2048Encrypt",
+    "Rsa2048Decrypt",
+    "Rsa2048Delete",
+    "Rsa3072Gen",
+    "Rsa3072Sign",
+    "Rsa3072Verify",
+    "Rsa3072Encrypt",
+    "Rsa3072Decrypt",
+    "Rsa3072Delete",
+    "Rsa4096Gen",
+    "Rsa4096Sign",
+    "Rsa4096Verify",
+    "Rsa4096Encrypt",
+    "Rsa4096Decrypt",
+    "Rsa4096Delete",
+    "SymmWrite",
+    "SymmEncryptOneShot",
+    "SymmDecryptOneShot",
+    "SymmEncryptCreate",
+    "SymmEncryptInit",
+    "SymmEncryptUpdate1",
+    "SymmEncryptUpdate2",
+    "SymmEncryptFinal",
+    "SymmEncryptDelete",
+    "SymmDecryptCreate",
+    "SymmDecryptInit",
+    "SymmDecryptUpdate1",
+    "SymmDecryptUpdate2",
+    "SymmDecryptFinal",
+    "SymmDecryptDelete",
+    "SymmDelete",
+    "MacWrite",
+    "MacSignOneShot",
+    "MacVerifyOneShot",
+    "MacSignCreate",
+    "MacSignInit",
+    "MacSignUpdate1",
+    "MacSignUpdate2",
+    "MacSignFinal",
+    "MacSignDelete",
+    "MacVerifyCreate",
+    "MacVerifyInit",
+    "MacVerifyUpdate1",
+    "MacVerifyUpdate2",
+    "MacVerifyFinal",
+    "MacVerifyDelete",
+    "MacDelete",
+    "AesSessionCreateKey",
+    "AesSessionCreateBinary",
+    "AesSessionCreateSession",
+    "AesSessionAuthenticate",
+    "AesSessionReadBinary",
+    "AesSessionUpdateKey",
+    "AesSessionCloseSession",
+    "AesSessionRecreateSession",
+    "AesSessionReAuthenticate",
+    "AesSessionReadBinary2",
+    "AesSessionDeleteBinary",
+    "AesSessionDeleteKey",
+    "Pbkdf2WritePin",
+    "Pbkdf2Calculate",
+    "Pbkdf2Compare",
+    "Pbkdf2DeletePin",
+    "ImportWrite",
+    "ImportCipher",
+    "ImportExport",
+    "ImportDelete",
+    "ImportDeletionWorked",
+    "ImportImport",
+    "ImportCipher2",
+    "ImportComp",
+    "ImportDeleteFinal",
+]
+
+
+@test_case("se050", "SE050")
+def test_se050(ctx: TestContext, device: Nitrokey3Base) -> TestResult:
+    from queue import Queue
+
+    if not isinstance(device, Nitrokey3Device):
+        return TestResult(TestStatus.SKIPPED)
+    firmware_version = ctx.firmware_version or device.version()
+    if (
+        firmware_version.core() < Version(1, 5, 0)
+        or firmware_version.core() >= Version(1, 6, 0)
+        or firmware_version.pre is None
+    ):
+        return TestResult(TestStatus.SKIPPED)
+
+    que: Queue[Optional[bytes]] = Queue()
+
+    def internal_se050_run(q: Queue[Optional[bytes]]) -> None:
+        q.put(AdminApp(device).se050_tests())
+
+    t = Thread(target=internal_se050_run, args=[que])
+    t.start()
+    bar = tqdm(
+        desc="Running SE050 test", unit="%", bar_format="{l_bar}{bar}", total=900
+    )
+    # 1m30 in increments of 0.1 second
+    for i in range(900):
+        t.join(0.1)
+        bar.update(1)
+        if not t.is_alive():
+            break
+        if i == 900 - 1:
```

Comment:
nit: use a constant for the timeout

Reply:
and maybe use `else` for the last occurence

Reply:
for-else is nice